src/server.js

```diff
@@ -91,6 +92,20 @@ function init(socket, client, token) {
 		socket.emit("auth");
 		socket.on("auth", auth);
 	} else {
+		var ip = socket.handshake.headers['x-forwarded-for'] || socket.request.connection.remoteAddress;
```

Comment:
> Yep. You are right... Looks like you need a list of trusted proxies. Or a simple toggle in configuration if you are behind a proxy or not.

The second solution (the toggle in configuration) seems simple and fine to me.

> I will remove that from this PR and maybe create another one to add that feature properly implemented again.

IMHO, better to stick on this very PR, reworking it. It's easier to continue conversation, and avoid polluting the bug tracker.